---
title: "Workflow_Yosia"
author: "Yosia"
date: "10/13/2019"
output: html_document
---
#Loading the data
I used the links from the class github to load the two needed data sets.

```{r}
download.file("https://raw.githubusercontent.com/EEOB-BioData/BCB546X-Fall2019/master/assignments/UNIX_Assignment/snp_position.txt", destfile = "./snp_position.txt")

download.file("https://raw.githubusercontent.com/EEOB-BioData/BCB546X-Fall2019/master/assignments/UNIX_Assignment/fang_et_al_genotypes.txt", destfile = "./fang_et_al_genotypes.txt")
```
#Loading tidyverse
I loaded the package tidyverse and read the data sets as "fang" & "snp"
```{r}
library(tidyverse)
fang <- read_tsv("./fang_et_al_genotypes.txt")

snp <- read_tsv("./snp_position.txt")
```

#Part 1
##Data inspection
Data inspection was carried out to establish number of rows, columns, inspected the dimensions of files by using the "dim" function and limit the entire file into a short piece of file (10 rows and 5 columns). Also i used the summary function on the data files to .
To see how many groups exist and what are their names,i used "unique(filename$variableofinterest)"
```{r}
#dim(fang)
#dim(snp)
```
```{r}
#fang[1:10, 1:5]
#snp[1:10, 1:5]
#length(fang$Sample_ID)
#str(fang)
#str(snp)
#summary (fang)
#summary(snp)
#unique(fang$Group)
```

##Part II
#Data processing
Data processing was done before doing visualization. 
Here i transposed and merged the entire fang_et_al_genotype.txt file with the snp_position.txt. From the fang_et_al_genotype we filtered out the 3 groups we want to work with for maize and also we did the same for teosinte where we filtered out the 3 groups we are intrested in working with. I choose the groups of interest and transfered them into a new file that i transposed by "t()"

I formatted the snp_position.txt data file such that i only select data in columns '"SNP_ID", "Chromosome", and "Position"

After the above manipulations i merged maize and teosinte transposed data with the snp formatted data

In both the maize and teosinte data i replaced "?" with "-/-" and i used the loop to create increasing and decreasing files for maize and teosinte

Next i transposed the entire fang_et_al_genotypes.txt to prepare it to merge with the entire snp_position.txt

```{r}
Maize_transposed <- filter(fang, Group %in% c("ZMMLR","ZMMMR","ZMMIL")) %>%
  select(-JG_OTU, -Group) %>% column_to_rownames(., var = "Sample_ID") %>% t() %>%
  as.data.frame() %>% rownames_to_column(., var = "SNP_ID")

Teosinte_transposed <- filter(fang, Group %in% c("ZMPBA","ZMPIL","ZMPJA"))%>%
  select(-JG_OTU, -Group) %>% column_to_rownames(., var = "Sample_ID") %>% t() %>%
  as.data.frame() %>% rownames_to_column(., var = "SNP_ID")

snp_formatted <- snp %>% select(SNP_ID, Chromosome, Position)

Maize_merged <- merge(snp_formatted, Maize_transposed ) %>%
  filter(., Position != "multiple",  Chromosome != "unknown")

Teosinte_merged <- merge(snp_formatted, Teosinte_transposed) %>%
  filter(., Position != "multiple",  Chromosome != "unknown")

Maize_replaced <- Maize_merged %>% as.matrix()
Teosinte_replaced <- Teosinte_merged %>% as.matrix()

Maize_replaced[Maize_replaced == "?/?"] <- "-/-"
Teosinte_replaced[Teosinte_replaced == "?/?"] <- "-/-"

Maize_replaced <- as.tibble(Maize_replaced)
Teosinte_replaced <- as.tibble(Teosinte_replaced)


for (i in 1:10) {
 
  m1 <- Maize_merged %>% filter(., Chromosome == i[1]) %>% arrange(.,as.numeric(Position))
  m2 <- Maize_replaced %>% filter(., Chromosome == i[1]) %>% arrange(.,desc(as.numeric(Position)))
 
  write_tsv(m1, paste("Maize_chr", i[1], "increasing.txt", sep = "_"))
  write_tsv(m2, paste("Maize_chr", i[1], "decreasing.txt", sep = "_"))
 
  t1 <- filter(Teosinte_merged, Chromosome == i[1]) %>% arrange(., as.numeric(Position))
  t2 <- filter(Teosinte_replaced, Chromosome == i[1]) %>% arrange(., desc(as.numeric(Position)))
 
  write_tsv(t1, paste("Teosinte_chr", i[1], "_increasing.txt", sep = "_"))
  write_tsv(t2, paste("Teosinte_chr", i[1], "_decreasing.txt", sep = "_"))
}

```
First i transposed fang original file and removed two columns that are not necessary (JG_OTU and Group). Now i have SNP ids in rows and sample names in columns. 
Then i converted the file into a matrix so we can annotate number 1 for homozygosity, zero for heterozygosity and NA for missing data that are given by question marks. 
i got a count the number of homozygotes, heterozygotes, and missing SNPs. 
Formatted sample is reshaping the data into a proper arrangment so we can plot them in stack bar graphs using ggplot. 

```{r}
Fang_transposed <- fang %>% select( -JG_OTU, -Group) %>% column_to_rownames(., var = "Sample_ID") %>% t() %>% as.data.frame() %>% rownames_to_column (., var = "SNP_ID")
Fang_merged <- merge (snp_formatted, Fang_transposed) %>% filter( ., Position != "mulitiple", Position !="unknown")

ggplot(Fang_merged, aes(x= as.character (Chromosome))) + geom_bar (aes(fill= as.character(Chromosome))) +theme_bw()

ggplot(Fang_merged, aes(x= as.numeric(Position))) + geom_density(aes(fill = as.character(Chromosome))) + facet_wrap(~Chromosome, nrow = 2, ncol = 5) + theme_bw()

```
We need to remove the column JG_OTU from the fang file. Then  we allocate the SNPs to each sample. So we have 939 SNPs and 2782 samples, hence we will have (939*2782=2734706) rows (about 900 SNPs for each sample). 
Next, we add a new column called new_sequence in that ,by using ifelse, it will contain the SNPs characters as homo/heterozygous, and missing for unavailable data. 
Also, we use position = "fill" to make each bar to fill the entire length of graph. In other words, we standardize the data so we can see how many SNP characters does each sample/group contain.

```{r}
tidy_fang<- fang %>% select(-JG_OTU) %>% gather(key = "SNP_ID", value = "Sequence", names(fang)[4:986])

tidy_fang <- tidy_fang %>%
  mutate(new_sequence = ifelse(Sequence %in% c("A/A","T/T","C/C","G/G"), "Homozygous",
  ifelse(Sequence == "?/?", "Missing","Heterozygous")))
```
##More Data Visualization

```{r}
ggplot(tidy_fang, aes(x = Sample_ID, fill = new_sequence)) +
  geom_bar(position = "fill") + theme_bw()
```
```{r}
ggplot(tidy_fang, aes(x = Group , fill = new_sequence)) + geom_bar(position = "fill") +
  theme_bw() + theme(axis.text.x = element_text(angle = 90))
```
#My own Visualization
In my own visualization i wanted to see what is the combination of SNPs and what fraction does each SNP occupy.

```{r}
ggplot(filter(tidy_fang, Sequence != "?/?") , aes(x = Sample_ID, fill = Sequence)) +
  geom_bar (position = "fill") +theme_bw()
```







